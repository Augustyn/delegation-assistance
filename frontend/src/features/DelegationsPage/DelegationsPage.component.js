import React, { Fragment } from "react";
import { bool } from "prop-types";

import LayoutMain from "../../components/layouts/LayoutMain/LayoutMain.container";
import Card from "../../components/Card/Card.component";
import ButtonLink from "../../components/ButtonLink/ButtonLink.component";

const DelegationsPage = props => {
  const { isManager, isEmployee } = props;
  return (
    <LayoutMain title="Delegations dashboard">
      <Fragment>
        <Card title="Statistics">
          {isManager ? (
            <div>
              <ButtonLink href="/delegations/manage" text="Manage delegations" />
            </div>
          ) : null}
          {isEmployee ? (
            <Fragment>
              <div>
                <ButtonLink href="/delegations/my" text="My delegations" />
              </div>
              <div>
                <ButtonLink href="/delegations/create" text="Create delegation" />
              </div>
            </Fragment>
          ) : null}
        </Card>
      </Fragment>
    </LayoutMain>
  );
};

DelegationsPage.propTypes = {
  isManager: bool,
  isEmployee: bool
};

export default DelegationsPage;
