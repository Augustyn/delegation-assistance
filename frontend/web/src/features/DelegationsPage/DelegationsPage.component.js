import React from "react";

import LayoutMain from "../../components/layouts/LayoutMain/LayoutMain.component";

const DelegationsPage = props => {
  return (
    <LayoutMain>
      <div>Delegations page</div>
      <a href="/">Back to home</a>
    </LayoutMain>
  );
};

export default DelegationsPage;
