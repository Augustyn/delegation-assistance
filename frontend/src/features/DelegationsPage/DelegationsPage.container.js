import React, { Component } from "react";
import { connect } from "react-redux";
import { array, bool, func } from "prop-types";

import DelegationsPage from "./DelegationsPage.component";
import { isUserEmployee } from "../../selectors/user.selectors";
import { fetchMyDelegations, fetchDelegations } from "../../actions/delegations.actions";
import {
  getDelegations,
  getRejectedDelegations,
  getWaitingDelegations,
  getDelegationFetching
} from "../../selectors/delegations.selectors";

class DelegationsPageContainer extends Component {
  static propTypes = {
    delegations: array,
    fetchDelegations: func,
    fetchMyDelegations: func,
    fetching: bool,
    isEmployee: bool,
    rejectedDelegations: array,
    waitingDelegations: array
  };

  componentDidMount() {
    if (this.props.isEmployee) {
      this.props.fetchMyDelegations();
    } else {
      this.props.fetchDelegations();
    }
  }

  render() {
    return (
      <DelegationsPage
        isEmployee={this.props.isEmployee}
        fetching={this.props.fetching}
        delegations={this.props.delegations}
        rejectedDelegations={this.props.rejectedDelegations}
        waitingDelegations={this.props.waitingDelegations}
      />
    );
  }
}

const mapStateToProps = state => ({
  isEmployee: isUserEmployee(state),
  fetching: getDelegationFetching(state),
  delegations: getDelegations(state),
  rejectedDelegations: getRejectedDelegations(state),
  waitingDelegations: getWaitingDelegations(state)
});

const mapDispatchToProps = {
  fetchDelegations,
  fetchMyDelegations
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DelegationsPageContainer);
