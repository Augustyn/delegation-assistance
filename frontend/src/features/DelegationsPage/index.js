import DelegationsPage from "./DelegationsPage.container";
export default DelegationsPage;
