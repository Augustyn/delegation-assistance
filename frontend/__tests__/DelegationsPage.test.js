import React from "react";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import DelegationsPage from "../src/features/DelegationsPage/DelegationsPage.component";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    handleSubmit: jest.fn(),
    subbmiting: false,
    invalid: false
  };

  const enzymeWrapper = shallow(<DelegationsPage {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Delegations Page", () => {
  let enzymeWrapper;
  let props;

  beforeEach(() => {
    ({ enzymeWrapper, props } = setup());
  });

  it("should render correctly", () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
