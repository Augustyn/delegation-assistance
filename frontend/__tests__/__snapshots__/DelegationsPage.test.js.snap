// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delegations Page should render correctly 1`] = `
<Connect(LayoutMainContainer)
  title="Delegations dashboard"
>
  <Card
    classes=""
    number={0}
    title="Statistics"
  />
</Connect(LayoutMainContainer)>
`;
